src/app/games/coerceo/tests/CoerceoRules.spec.ts

```diff
@@ -220,18 +262,92 @@ describe('CoerceoRules', () => {
                 [N, N, N, N, N, N, X, _, X, _, _, _, N, N, N],
                 [N, N, N, N, N, N, _, _, _, N, N, N, N, N, N],
             ];
-            const state: CoerceoState = new CoerceoState(board, 1, PlayerNumberMap.of(0, 0), PlayerNumberMap.of(0, 0));
-            const move: CoerceoMove = movement(new Coord(8, 6), CoerceoStep.DOWN_RIGHT);
             const expectedState: CoerceoState =
                 new CoerceoState(expectedBoard, 2, PlayerNumberMap.of(0, 1), PlayerNumberMap.of(0, 1));
             RulesUtils.expectMoveSuccess(rules, state, move, expectedState, defaultConfig);
         });
 
+        it('should throw when trying out of range starting coord', () => {
+            // Given any board
+            const state: CoerceoState = rules.getInitialState(defaultConfig);
+
+            // When trying a move starting from out of range
+            const move: CoerceoMove = CoerceoRegularMove.ofMovement(new Coord(-1, 0), CoerceoStep.LEFT);
+
+            // Then it should throw
+            const reason: string = 'Accessing coord not on board (-1, 0).';
+            function tryingOutOfRangeStartingMove(): void {
+                RulesUtils.expectMoveFailure(rules, state, move, reason, defaultConfig);
+            }
+            TestUtils.expectToThrowAndLog(tryingOutOfRangeStartingMove, reason);
+        });
+
+        it('should not allow starting coord unreachable', () => {
+            // Given any board
+            const state: CoerceoState = rules.getInitialState(defaultConfig);
+
+            // When starting a move on an unreachable space
+            const move: CoerceoMove = CoerceoRegularMove.ofMovement(new Coord(0, 0), CoerceoStep.LEFT);
+
+            // Then it should throw
+            const reason: string = 'Cannot start with a coord outside the board (0, 0).';
+            function allowOutOfRangeLandingCoord(): void {
+                RulesUtils.expectMoveFailure(rules, state, move, reason, defaultConfig);
+            }
+            TestUtils.expectToThrowAndLog(allowOutOfRangeLandingCoord, reason);
+        });
+
+        it('should not allow starting coord out of board', () => {
```

Comment:
quelle différence entre celui-ci et le "should throw when trying out of range starting coord"?

Reply:
comme dirais Calogrenant "ah rien du tout"